pkg/kubelet/lifecycle/handlers.go

```diff
@@ -99,6 +99,9 @@ func resolvePort(portReference intstr.IntOrString, container *v1.Container) (int
 	}
 	for _, portSpec := range container.Ports {
 		if portSpec.Name == portName {
+			if protocol != "" && portSpec.Protocol != protocol {
```

Comment:
a comment here to call out that container port names are unique across protocols would be helpful: https://github.com/kubernetes/kubernetes/pull/99431#discussion_r586994076